tests/checkers/resource_files.py

```diff
@@ -61,3 +63,34 @@ def check_resource_file_name(file_path):
 def check_file_extensions(file, lines, options=None):
     """ Check that there is no file without extension. """
     yield 0, "the file does not have an extension"
+
+@sphinxlint.checker('')
+def check_is_resource_file_used(file, options=None):
+    """ Check each resource file is used in it's .rst file. """
+    resource_file = os.path.basename(file)
+    resource_folder = os.path.dirname(file)
+    rst_file = resource_folder + '.rst'
+    image_pattern = re.compile(r'(\.\. (image|figure)::\s*)([^\s]+)')
+    if os.path.exists(rst_file):
+        rst_file_contents = open(rst_file)
```

Comment:
Except for the line with the RE pattern, all of this should use `pathlib`'s `Path`, not `os`.